plugins/catalog-react/src/apis/EntityPresentationApi/defaultEntityPresentation.ts

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  CompoundEntityRef,
+  DEFAULT_NAMESPACE,
+  Entity,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import { IconComponent } from '@backstage/core-plugin-api';
+import ApartmentIcon from '@material-ui/icons/Apartment';
+import BusinessIcon from '@material-ui/icons/Business';
+import ExtensionIcon from '@material-ui/icons/Extension';
+import HelpIcon from '@material-ui/icons/Help';
+import LibraryAddIcon from '@material-ui/icons/LibraryAdd';
+import LocationOnIcon from '@material-ui/icons/LocationOn';
+import MemoryIcon from '@material-ui/icons/Memory';
+import PeopleIcon from '@material-ui/icons/People';
+import PersonIcon from '@material-ui/icons/Person';
+import get from 'lodash/get';
+import { EntityRefPresentationSnapshot } from './EntityPresentationApi';
+
+const UNKNOWN_KIND_ICON: IconComponent = HelpIcon;
+
+const DEFAULT_ICONS: Record<string, IconComponent> = {
+  api: ExtensionIcon,
+  component: MemoryIcon,
+  system: BusinessIcon,
+  domain: ApartmentIcon,
+  location: LocationOnIcon,
```

Comment:
Agreed. But since this is a service, there's no convenient path to always consistently get at the app icons. Injecting a renderer of your own becomes pretty difficult since you cant `useApp` at that point (you are outside of React).

I _could_ make it so that `useEntityPresentation` did the injection always, but that leaks through several interface layers then and is pretty unfortunate.

I'd prefer to instead in a separate PR add an `appIconsApiRef` similar to the theme and translation refs. Then, we can more easily hide all of this without affecting any API surfaces.

Reply:
Sounds like a plan, agreed!